src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,371 @@
+    end
+    
+end
+const BinomialHeap{T} = MutableBinomialHeap{T}
+BinomialHeap(xs::AbstractVector{T}) where T = MutableBinomialHeap{T}(xs)
+
+struct HeapBoundsError <: Exception
+    msg::AbstractString
+end
+
+function show(io::IO, _heap::MutableBinomialHeap{T}) where T
+    print(io, "MutableBinomialHeap")
+    list = _heap.rootList
+    n = length(list)
+    if n == 0
+        print(io, "{$(T)}")
+    end
+    print(io, "(")
+    for i = 1:n
+        print_tree(io, list[i])
+    end
+    print(io, ")")
+end
+
+function print_tree(io::IO, h::Union{MutableBinomialHeapNode{T},Nothing}) where T
+    while h !== nothing 
+        print(io, h.data)
+        print(io, ",")
+        print_tree(io, h.child) 
+        h = h.sibling; 
+    end
+end
+#################################################
+#
+#   core implementation
+#
+#################################################
+# make Binomial_heap from Vector
+function _make_binomial_heap(xs::AbstractVector{T}, Typ::Type{T}) where T
+    BinomialHeap = MutableBinomialHeap{Typ}()
+    for x in xs
+        push!(BinomialHeap, x)
+    end
+    BinomialHeap
+end
+
+# This function merge two Binomial Trees. 
+function merge_binomial_trees!(b1::MutableBinomialHeapNode{T}, b2::MutableBinomialHeapNode{T}) where T 
+    if b1.data > b2.data
+        b1, b2 = b2, b1
+    end
+    b2.parent = b1 
+    b2.sibling = b1.child 
+    b1.child = b2 
+    b1.degree += 1 
+    return b1
+end
+# adjust! function rearranges the heap so that 
+# heap is in increasing order of degree and 
+# no two binomial trees have same degree in this heap 
+function adjust!(heap::MutableBinomialHeap{T}) where T
+    _heap = heap.rootList
+    n = length(_heap)
+    if n <= 1
+        return
+    end
+    i = 1
+    if n == 2
+        j = 2
+        k = n + 1
+    else
+        j = i + 1
+        k = j + 1
+    end
+    while i <= length(_heap)
+        current_size = length(_heap) + 1
+        if j >= current_size
+            i += 1
+        elseif _heap[i].degree < _heap[j].degree
+            i += 1
+            j += 1
+            if k !== current_size
+                k += 1
+            end
+        elseif (k < current_size)  && (_heap[i].degree == _heap[j].degree) && (_heap[i].degree == _heap[k].degree)
+            i += 1
+            j += 1
+            k += 1  
+        elseif _heap[i].degree == _heap[j].degree
+            temp = merge_binomial_trees!(_heap[i], _heap[j])
+            _heap[i] = temp
+            @inbounds _heap[j] = nothing
+            filter!(e->e ≠ nothing, _heap)
+            if k < (length(_heap) + 1)
+                k = j + 1 
+            end
+        end
+    end
+end
+
+function insert_tree_in_heap!(_heap::MutableBinomialHeap{T}, tree::MutableBinomialHeapNode{T}) where T 
+    temp = MutableBinomialHeap{T}()
+    push!(temp.rootList, tree)
+    _union!(_heap, temp, true)
+    adjust!(_heap)
+    return
+end   
+function remove_min__from_tree!(tree::MutableBinomialHeapNode{T}) where T 
```

Comment:
```suggestion

function remove_min_from_tree!(tree::MutableBinomialHeapNode{T}) where T 
```

Seemed to have a double `_` and also mised the line break between functions